internal/groups/api/endpoints.go

```diff
@@ -284,3 +294,25 @@ func buildGroupsResponse(gp groups.Page, filterByID bool) groupPageRes {
 
 	return res
 }
+
+func buildChannelsResponse(cp groups.Page, filterByID bool) channelPageRes {
+	res := channelPageRes{
+		pageRes: pageRes{
+			Total: cp.Total,
+			Level: cp.Level,
+		},
+		Channels: []viewGroupRes{},
+	}
+
+	for _, channel := range cp.Groups {
+		view := viewGroupRes{
+			Group: channel,
+		}
+		if filterByID && channel.Level == 0 {
+			continue
+		}
```

Comment:
Move this to the top (to avoid unnecessary allocations) or use `if !filterByID && channel.Level != nil {...}`